console/src/components/Services/Events/AdhocEvents/Add/Add.tsx

```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import DateTimePicker from 'react-datetime';
+import { Moment } from 'moment';
+import { useAdhocEventAdd } from './state';
+import styles from '../../Events.scss';
+import Button from '../../../../Common/Button/Button';
+import { Dispatch } from '../../../../../types';
+import AceEditor from '../../../../Common/AceEditor/BaseEditor';
+import Headers from '../../../../Common/Headers/Headers';
+import RetryConfEditor from '../../Common/Components/RetryConfEditor';
+import Tooltip from '../../../../Common/Tooltip/Tooltip';
+import CollapsibleToggle from '../../../../Common/CollapsibleToggle/CollapsibleToggle';
+import { createScheduledEvent } from '../../ServerIO';
+
+type Props = {
+  dispatch: Dispatch;
+};
+
+const Add: React.FC<Props> = ({ dispatch }) => {
+  const { state, setState } = useAdhocEventAdd();
+  const {
+    webhook,
+    time,
+    payload,
+    headers,
+    retryConf,
+    comment,
+    loading,
+  } = state;
+
+  const setWebhookValue = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setState.webhook(e.target.value);
+  const setTimeValue = (e: Moment | string) => {
+    if (typeof e !== 'string') {
+      setState.time(e.toDate());
+    }
+  };
+  const setComment = (e: React.ChangeEvent<HTMLInputElement>) =>
+    setState.comment(e.target.value);
+
+  const sectionize = (section: JSX.Element) => (
```

Comment:
Because there need to be proper UI components to write higher order UI components. I think we had settled on making higher order UI components after the basic UI components get merged. I was planning to write components then.

But since we have a lot of time right now, I have broken this up into components. This will require rewrite later when we build higher order UI components though.

Reply:
It's not a higher-order component. It's just a component. That takes children and does nothing. It doesn't provide additional props. It doesn't add any logic. I don't see why it can't be for example:
```tsx
const Section: React.FC => ({children}) => (
   <div className={styles.add_mar_bottom}>
      {children}
      <hr />
    </div>
)
```